### PR TITLE
[doc] Add information regarding header include order

### LINKED ISSUE #3904: [docs] Ordering of header includes
**Description**

There is no documentation available to refer, regarding the ordering of includes to be followed throughout the source.

**Context**

While running clang-format, it rearranges header includes:
```
#include <pcl/common/eigen.h>	          #include <ctime>
#include <crime>	            ->    #include <pcl/common/eigen.h>	
#include <pcl/tracking/tracker.h>	  #include <pcl/tracking/tracker.h>	
```
[Blame from my PR](https://github.com/PointCloudLibrary/pcl/blame/723f2886a442dc82034be77568a8c65e913f9977/tracking/include/pcl/tracking/impl/tracker.hpp#L4L6)

While searching for documentation regarding the ordering, I found out there isn't one.

**Expected behaviour**

- A set pattern to be enforced throughout the source, with exceptions, if any, documented.
- No rearrangement while formatting code.

**Describe the solution you'd like**

1. Documentation of the order of includes, if we have one. Else, adoption of one.
    Example: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes
2. Addition of a style option in our `.clang-format` file, to prevent rearranging.
    - Set `SortIncludes (bool)` as `false`.
    - Alternatively, explore options like `IncludeBlocks (IncludeBlocksStyle)` and `IncludeCategories (std::vector<IncludeCategory>)`.
    - [Clang format 10 documentation](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormatStyleOptions.html)
3. Enforcing the order throughout the source (Discussion needed).
***
Initially suggested by @kunaltyagi in https://github.com/PointCloudLibrary/pcl/pull/3814#discussion_r406749664

### PR DESCRIPTION
**Pull request by divmadan**
closes #3904

### CODE REVIEW

File: doc/advanced/content/pcl_style_guide.rst [hunk 1]
```diff
@@ -323,6 +324,36 @@ Command line usage:
    $ make format
 
 
+2.7. Includes
+^^^^^^^^^^^^^
+
+For consistent usage, headers should be included in the following order with alphabetical grouping ensured:
+
+1. PCL headers
+  i. All modular PCL includes, except main includes of common module
```
Comment (kunaltyagi) [inline doc/advanced/content/pcl_style_guide.rst:333]: GitHub preview is bad here

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: Taken care of.

**Comment by divmadan** [comment]
Webpage looks fine :+1: 
Please squash while merging.

File: doc/advanced/content/pcl_style_guide.rst [hunk 2]
```diff
@@ -323,6 +323,40 @@ Command line usage:
...
+2.  Major 3rd-Party components of tests and modules
+
+    i.  gtest
+    #.  boost
+    #.  Eigen
+    #.  flann
+3.  Major 3rd-Party components of apps
+
+    i.  Qt
+    #.  ui\_
```
Comment (SunBlack) [inline doc/advanced/content/pcl_style_guide.rst:344]:
```suggestion
    #.  ui-files
```

File: doc/advanced/content/pcl_style_guide.rst [hunk 3]
```diff
@@ -323,6 +323,40 @@ Command line usage:
...
+    #.  vtk
+4.  Minor 3rd-Party components
+
+    i.  librealsense
+    #.  ros/message_filters
+    #.  opencv(2)
```
Comment (SunBlack) [inline doc/advanced/content/pcl_style_guide.rst:350]:
```suggestion
    #.  opencv/opencv2
```

File: doc/advanced/content/pcl_style_guide.rst [hunk 4]
```diff
@@ -323,6 +323,40 @@ Command line usage:
...
+    #.  ui-files
+    #.  vtk
+4.  Minor 3rd-Party components
+
+    i.  librealsense
+    #.  ros/message_filters
+    #.  opencv/opencv2
+    #.  tide
+    #.  thrust
+    #.  OpenGL(UT)/GL(UT)
```
Comment (SunBlack) [inline doc/advanced/content/pcl_style_guide.rst:353]: I missed this (the user doesn't know you copy & pasted regular expression code ;-) ):
```suggestion
    #.  OpenGL, GL & GLUT
```

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: I didn't copy. I just thought it was understood. :laughing:

**Comment by kunaltyagi** [comment]
`doc` succeeded. Others cancelled manually

File: doc/advanced/content/pcl_style_guide.rst [hunk 5]
```diff
@@ -323,6 +323,40 @@ Command line usage:
...
 
+2.7. Includes
+^^^^^^^^^^^^^
+
+For consistent usage, headers should be included in the following order with alphabetical grouping ensured:
+
+1.  PCL headers
+
+    i.  All modular PCL includes, except main includes of common module
+    #.  The main PCL includes of common module
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:334]: The concept of what is a "main include of common module" is not clear. I had to go to the clang-format file to understand what it represented. Provide examples for what a "normal modular include" and a "main include of a common module" are.

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: Looks good? 1f52a71

**Comment by divmadan** [comment]
Do squash before merging :relaxed:

File: doc/advanced/content/pcl_style_guide.rst [hunk 6]
```diff
@@ -323,6 +323,70 @@ Command line usage:
...
+
+           #include <pcl/common/common.h>
+
+        .. code-block:: cpp
+  
+           #include <pcl/simulation/camera.h>
+
+        .. code-block:: cpp
+
+           #include <pcl/ml/dt/decision_forest.h>
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:347]:
```suggestion
           #include <pcl/common/common.h>
           #include <pcl/simulation/camera.h>
           #include <pcl/ml/dt/decision_forest.h>
```

File: doc/advanced/content/pcl_style_guide.rst [hunk 7]
```diff
@@ -323,6 +323,70 @@ Command line usage:
...
+
+           #include <pcl/memory.h>
+
+        .. code-block:: cpp
+  
+           #include <pcl/pcl_macros.h>
+
+        .. code-block:: cpp
+
+           #include <pcl/point_cloud.h>
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:363]: same for this

Comment (divmadan) [reply doc/advanced/content/pcl_style_guide.rst]: Done.

**Pull request merged**